client/auth/controller.js

```diff
@@ -109,5 +114,49 @@ module.exports = {
 				window.location = '/';
 			}
 		} );
+	},
+
+	magicLoginRequestEmailForm: function( context ) {
+		renderWithReduxStore( (
+				<Main className="auth__magic-login-request-form">
+					<RequestLoginEmailForm />
+				</Main>
+			),
+			document.getElementById( 'primary' ),
+			context.store
+		);
+	},
+
+	magicLoginLinkWasSent: function( context ) {
+		renderWithReduxStore( (
+				<Main className="auth__magic-link-was-sent">
+					<EmailedLoginLinkSuccessfully />
+				</Main>
+			),
+			document.getElementById( 'primary' ),
+			context.store
+		);
+	},
+
+	magicLoginClickHandler: function( context ) {
+		renderWithReduxStore( (
+				<Main className="auth__magic-login-handle-click">
+					<HandleEmailedLinkForm />
+				</Main>
+			),
+			document.getElementById( 'primary' ),
+			context.store
+		);
+	},
+
+	magicLoginHasExpired: function( context ) {
+		renderWithReduxStore( (
+				<Main className="auth__magic-link-exipred">
+					<EmailedLoginLinkExpired />
+				</Main>
+			),
+			document.getElementById( 'primary' ),
+			context.store
+		);
```

Comment:
Minor: there's a `renderPage` helper which makes this shorter but also untangles the controller from DOM calls.